src/model/model.ts

```diff
@@ -129,116 +159,119 @@ export abstract class AbstractModel<
       ...mergedContext,
     });
 
-    return Sentry.startSpan({ name: spanName, op: 'llm' }, async (span) => {
-      const start = Date.now();
-
-      // Add tags for the prompt name and version if present
-      Sentry.setTags({
-        ...(typeof mergedContext?.promptName === 'string'
-          ? { promptName: mergedContext.promptName }
-          : {}),
-        ...(typeof mergedContext?.promptVersion === 'string'
-          ? { promptVersion: Number(mergedContext.promptVersion) }
-          : {}),
-      });
-
-      span.setAttributes({
-        ...extractAttrsFromContext(mergedContext),
-        ...extractAttrsFromParams({
-          modelType: this.modelType,
-          modelProvider: this.modelProvider,
-          ...mergedParams,
-        }),
-      });
-
-      await Promise.allSettled(
-        this.events.onStart?.map((event) =>
-          Promise.resolve(
-            event({
-              timestamp: new Date().toISOString(),
-              modelType: this.modelType,
-              modelProvider: this.modelProvider,
-              params: mergedParams,
-              context: mergedContext,
-            })
-          )
-        ) ?? []
-      );
-
-      const cacheKey = await this.cacheKey(mergedParams);
-
-      try {
-        // Check the cache
-        const cachedResponse =
-          this.cache && (await Promise.resolve(this.cache.get(cacheKey)));
-        if (cachedResponse) {
-          const response: MResponse = {
-            ...cachedResponse,
-            cached: true,
-            cost: 0,
-            latency: Date.now() - start,
-          };
-          span.setAttributes(extractAttrsFromResponse(response));
-          await Promise.allSettled(
-            this.events.onComplete?.map((event) =>
-              Promise.resolve(
-                event({
-                  timestamp: new Date().toISOString(),
-                  modelType: this.modelType,
-                  modelProvider: this.modelProvider,
-                  params: mergedParams,
-                  response,
-                  context: mergedContext,
-                  cached: true,
-                })
-              )
-            ) ?? []
-          );
-          return response;
-        }
+    return this.telemetry.startSpan(
```

Comment:
the only reason this block looks like a bigger change than it is is due to spacing. it's just replacing `Sentry` with `this.telemetry`.